core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+                        ng-click="onLoginButtonClicked('{{login_url}}')">
+                  <span translate="I18N_TOPNAV_SIGN_IN"></span>
+                  <span class="caret"></span>
+                </button>
+              </div>
+              <ul class="dropdown-menu oppia-navbar-dropdown" role="menu" style="margin-right: 15px; padding: 0;" ng-mouseover="onMouseoverDropdownMenu($event)" ng-mouseleave="onMouseoutDropdownMenu($event)">
+                <li>
+                  <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked('{{login_url}}')">
+                    <img ng-src="<[getStaticImageUrl('/signin/Red-signin-Long-base-44dp.png')]>">
+                  </a>
+                </li>
+              </ul>
+            </li>
+          {% endif %}
+        </ul>
+        <ul class="nav oppia-navbar-nav" ng-if="['create', 'explore', 'collection'].indexOf(NAV_MODE) === -1">
+          <ul ng-if="windowIsNarrow" class="nav oppia-navbar-tabs-narrow">
+            <create-activity-button></create-activity-button>
+          </ul>
+          <ul ng-if="!windowIsNarrow" class="nav oppia-navbar-tabs">
+            <create-activity-button></create-activity-button>
+            <li>
+              <div class="pull-right oppia-navbar-button-container">
+                <a href="/donate"
+                   class="btn oppia-navbar-button oppia-navbar-hide-on-small-width protractor-test-create-activity oppia-transition-200"
+                  <span translate="I18N_TOPNAV_DONATE"></span>
+                </a>
+              </div>
+            </li>
+
+            <li class="dropdown oppia-navbar-clickable-dropdown pull-right protractor-test-about-oppia-list-item">
+              <a class="oppia-navbar-tab">
+                <span translate="I18N_TOPNAV_ABOUT"></span>
+                <span class="caret"></span>
+              </a>
+              <ul class="dropdown-menu oppia-navbar-dropdown"
+                  ng-mouseover="onMouseoverDropdownMenu($event)"
+                  ng-mouseleave="onMouseoutDropdownMenu($event)">
+                <li><a class="protractor-test-about-link" href="/about" translate="I18N_TOPNAV_ABOUT_OPPIA"></a></li>
+                <li><a class="protractor-test-teach-link" href="/teach" translate="I18N_TOPNAV_TEACH_WITH_OPPIA"></a></li>
+                <li><a href="/blog" translate="I18N_TOPNAV_BLOG"></a></li>
+                <li><a href="/forum" translate="I18N_TOPNAV_FORUM"></a></li>
+                <li><a class="protractor-test-contact-link" href="/contact" translate="I18N_TOPNAV_CONTACT_US"></a></li>
+              </ul>
+            </li>
+            <li class="oppia-clickable-navbar-element pull-right">
+              <a class="oppia-navbar-tab" href="/library" translate="I18N_TOPNAV_LIBRARY"></a>
+            </li>
+          </ul>
+        </ul>
+      </div>
+
+      <div class="collapse navbar-collapse ng-cloak">
+        {% block navbar_breadcrumb %}
```

Comment:
We ought to figure out an Angular replacement for this functionality. I had a comment earlier for adding support for custom navbar options; it's needed to avoid some workarounds in the corresponding JS file for this directive. I get the sense a custom navbar breadcrumb wouldn't be too complicated, either. We could programmatically add directives at runtime; that's essentially what Jinja is doing.